### PR TITLE
Add a default directory for -Zdump-mir-dir

### PR DESCRIPTION
**Pull request by varkor**
The current behaviour of dumping in the current directory is rarely
desirable: a sensible default directory for dumping is much more
convenient. This makes sets the default value for `-Zdump-mir-dir`
to `mir_dump/`.

r? @eddyb

### CODE REVIEW

File: src/librustc/session/config.rs [hunk 1]
```diff
@@ -1172,7 +1172,7 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
           "emit noalias metadata for mutable references"),
     dump_mir: Option<String> = (None, parse_opt_string, [UNTRACKED],
           "dump MIR state at various points in translation"),
-    dump_mir_dir: Option<String> = (None, parse_opt_string, [UNTRACKED],
+    dump_mir_dir: Option<String> = (Some(String::from("mir_dump")), parse_opt_string, [UNTRACKED],
```
Comment (kennytm) [inline src/librustc/session/config.rs]: Could this be a `String` instead of `Option<String>` then?

Comment (varkor) [reply src/librustc/session/config.rs]: That's a good point! It's now non-optional.

**Pull request merged**